l10n_br_account_payment_order/wizards/account_move_line_change.xml

```diff
@@ -17,6 +17,10 @@
                         name="date_maturity"
                         attrs="{'invisible': [('change_type', '!=', 'change_date_maturity')], 'required': [('change_type', '=', 'change_date_maturity')]}"
                     />
+                    <field
+                        name="date"
+                        attrs="{'invisible': [('change_type', '!=', 'change_discount_date')], 'required': [('change_type', '=', 'change_discount_date')]}"
```

Comment:
Se o campo **date** passar a ser usado para os dois casos o invisible precisa ser um
('change_type', 'not in', ('change_date_maturity', 'change_discount_date')

Se for possível